src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -1271,6 +1274,44 @@ private void cleanupChatMessages()
         }
     }
 
+    @Override
+    public void setCustomNameTag(final String name)
+    {
+        if(citizenData != null && name != null && !citizenData.getName().equals(name))
+        {
+            if(Configurations.gameplay.allowGlobalNameChanges >= 0)
+            {
+                if (Configurations.gameplay.allowGlobalNameChanges == 0 &&
+                        Arrays.stream(Configurations.gameplay.specialPermGroup).noneMatch(owner -> owner.equals(colony.getPermissions().getOwnerName())))
+                {
+                    LanguageHandler.sendPlayersMessage(colony.getMessageEntityPlayers(), "com.minecolonies.coremod.citizen.rename.notAllowed");
+                    return;
+                }
+
+
+                if (colony != null)
+                {
+                    for (final CitizenData citizen : colony.getCitizens().values())
+                    {
+                        if (citizen.getName().equals(name))
+                        {
+                            LanguageHandler.sendPlayersMessage(colony.getMessageEntityPlayers(), "com.minecolonies.coremod.citizen.rename.same");
+                            return;
+                        }
+                    }
+                    this.citizenData.setName(name);
+                    this.citizenData.markDirty();
+                    super.setCustomNameTag(name);
+                }
+            }
+            else
+            {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS00108)